ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -471,6 +472,11 @@ - (NSString *)localizedInvalidValueStringWithAnswerString:(NSString *)text {
     return nil;
 }
 
+- (NSString *)stringForAnswer:(id)answer {
+    ORKAnswerFormat *impliedFormat = [self impliedAnswerFormat];
+    return impliedFormat == self ? nil : [impliedFormat stringForAnswer:answer];
+}
+
```

Comment:
I think we should aim for single returns so I'll change `isAnswerValid` and `isAnswerValidWithString:`